### PR TITLE
Fix response status code

### PR DESCRIPTION
**Pull request by liran-funaro**
Fix response status code to unauthorized when the request is not signed properly

Signed-off-by: Liran Funaro <liran.funaro@gmail.com>

### CODE REVIEW

File: internal/httphandler/data_request_handler.go [hunk 1]
```diff
@@ -176,7 +176,7 @@ func (d *dataRequestHandler) dataTransaction(response http.ResponseWriter, reque
 	}
 	if len(notSigned) > 0 {
 		sort.Strings(notSigned)
-		utils.SendHTTPResponse(response, http.StatusBadRequest,
+		utils.SendHTTPResponse(response, http.StatusUnauthorized,
```
Comment (tock-ibm) [inline internal/httphandler/data_request_handler.go:179]: Can you add a unit test that covers that?

Comment (liran-funaro) [reply internal/httphandler/data_request_handler.go:179]: There was a unit test that required the error code to be "bad request". I'm not sure why. And for some reason, the error code validation was commented out in the tests. I fixed the test and uncommented this check.

**Comment by tock-ibm** [comment]
@liran-funaro please rebase

**Comment by liran-funaro** [comment]
@tock-ibm rebased

**Review by tock-ibm — APPROVED** [review]

**Pull request merged**